tests/models/test_utils.py

```diff
@@ -161,3 +163,21 @@ def test_adding_libraries_to_model_when_version_source_None(sklearn_knn_model):
         assert wheeled_model_info.run_id is not None
         assert wheeled_model_info.run_id != original_run_id
         mlflow_client_mock.assert_called_once_with(model_name, "1")
+
+
+@pytest.mark.parametrize("orient", ["list", "records"])
+def test_schema_enforcement(orient):
+    # test inputs style signature
+    test_signature = {
+        "inputs": '[{"name": "a", "type": "long"}, {"name": "b", "type": "string"}]',
+        "outputs": '[{"name": "response", "type": "string"}]',
+    }
+    signature = ModelSignature.from_dict(test_signature)
+
+    data = {"a": [4, 5, 6], "b": ["a", "b", "c"]}
+
+    _enforce_schema(data, signature.inputs)
+
+    pd_data = pd.DataFrame(data)
+
+    _enforce_schema(pd_data.to_dict(orient=orient), signature.inputs)
```

Comment:
Can we do an assert on the types returned by `_enforce_schema` to ensure that they're the same as they were in MLflow 2.2? (We'll need to install 2.2 and test what the output of these calls would have been)

Reply:
Added a test based on offline validation of the logic pre 2.3 (after removing that silly numpy.size ==1 check (validated that it fails any type of schema enforcement checks with legitimate data structures passed in with "inputs" type)) in the new test suite for schema enforcement.